third_party/ibis/ibis_oracle/client.py

```diff
@@ -242,3 +244,36 @@ def list_tables(self, like=None, database=None, schema=None):
 
     def get_schema(self, name, schema=None):
         return self.table(name, schema=schema).schema()
+    
+    def sql(self, query):
+        """
+        Convert a Oracle query to an Ibis table expression
+
+        Parameters
+        ----------
+        query: string
+           SQL query to execute on connection
+
+        Returns
+        -------
+        table : TableExpr
+        """
+        limited_query = "SELECT * FROM ({}) t0 WHERE ROWNUM <= 1".format(query)
+        schema = self._get_schema_using_query(limited_query)
+        return ops.SQLQueryResult(query, schema, self).to_expr()
+
+    def _get_schema_using_query(self, limited_query):
+        type_map = {
+            "DB_TYPE_NUMBER": "int64",
```

Comment:
Should this be 'float64' since NUMBER can include precision/scale?

Reply:
The custom query validation fails if we use float64, because of same precision issue that we were facing in normal row validation.

Reply:
[Here](https://cx-oracle.readthedocs.io/en/latest/user_guide/sql_execution.html#fetch-data-types) all the supported data types in cx_oracle. 

Based on [this](https://cx-oracle.readthedocs.io/en/latest/api_manual/cursor.html#Cursor.description) Cursor description should include precision/scale. So we could use that to check whether to map NUMBER to float64 or int64.
```
for row in cur.proxy._cursor_description():
      precision, scale = row[4], row[5]
```